core/core-workflow-steps/src/main/java/org/eclipse/sw360/antenna/workflow/processors/LicenseKnowledgeBaseResolver.java

```diff
@@ -45,104 +44,40 @@ public LicenseKnowledgeBaseResolver(ILicenseManagementKnowledgeBase knowledgeBas
-     * LicenseKnowledgeBase.
-     *
-     * @param licenses
-     *            List of licenses for which the text will be set
-     */
-    private void setText(List<License> licenses) {
-        licenses.forEach(license -> {
-            String configuredText = license.getText();
-            if (StringUtils.isEmpty(configuredText)) {
-                String id = license.getName();
-                Optional.ofNullable(knowledgeBase.getTextForId(id))
-                    .ifPresent(license::setText);
-            }
-        });
-    }
-
-    private void setThreatGroup(List<License> licenses) {
-        licenses.forEach(license -> {
-            if (license.getThreatGroup() == null) {
-                LicenseThreatGroup threatGroup = this.knowledgeBase.getThreatGroupForId(license.getName());
-                license.setThreatGroup(threatGroup);
-            }
-        });
+    public LicenseKnowledgeBaseResolver(ILicenseManagementKnowledgeBase knowledgeBase,
+                                        Supplier<List<ILicenseManagementKnowledgeBase>> supplier) {
+        this.knowledgeBase = knowledgeBase;
+        this.knowledgeBaseSupplier = supplier;
     }
 
-    private void setClassification(List<License> licenses) {
-        licenses.forEach(license -> {
-            if (license.getClassification() == null) {
-                LicenseClassification licenseClassification = this.knowledgeBase.getClassificationById(license.getName());
-                license.setClassification(licenseClassification);                
-            }
-        });
+    public ILicenseManagementKnowledgeBase getKnowledgeBase() {
+        return this.knowledgeBase;
     }
 
     @Override
     public Collection<Artifact> process(Collection<Artifact> artifacts) {
-        LOGGER.debug("Resolve knowledge base...");
-        resolveKnowledgeBase(artifacts);
-        LOGGER.debug("Resolve knowledge base... done");
-        return artifacts;
+        return new LicenseKnowledgeBaseResolverImpl(knowledgeBase).resolveKnowledgeBase(artifacts);
     }
 
     @Override
     public void configure(Map<String,String> configMap) {
         super.configure(configMap);
-        this.knowledgeBase = new LicenseKnowledgeBaseFactory(context).get();
+        this.chosenManager = getConfigValue(CHOSEN_LICENSE_MANAGER_KEY, configMap);
+
+        if (knowledgeBaseSupplier == null) {
+            knowledgeBaseSupplier = new LicenseKnowledgeBaseFactory(context);
+        }
+
+        List<ILicenseManagementKnowledgeBase> licenseManagers = knowledgeBaseSupplier.get();
+
+        this.knowledgeBase = Optional.ofNullable(knowledgeBase)
+                .orElseGet(() -> licenseManagers.stream()
+                        .filter(l -> chosenManager.equals(l.getId()))
+                        .findFirst()
+                        .orElseGet(() -> licenseManagers.stream()
+                                .sorted(Comparator.comparing(ILicenseManagementKnowledgeBase::getPriority).reversed())
```

Comment:
The idea is that the CSV knowledgebase should check whether it can provide information, i.e., whether there is a csv file giving license information. If this is not the case, it should not be used as license knowledgebase. So Antenna chooses the knowledge bases according to the priority which is csv knowledgebase first and then the SPDX knowledgebase, but without a csv configuration the csv knowledgebase has to indicate somehow, that it cannot be used, so here the natural choice would be the SPDX knowledgebase. My proposal is that the getPriority method returns a value < 0 to indicate that the knowledgebase cannot be used.